File: Go/sum-of-digits-of-string-after-convert.go
func getLucky(s string, k int) int {
    sum := 0
    for _, s := range s {
        value := int(s - 'a' + 1)
        sum += value % 10 + value / 10
    }
    for k > 1 {
        newSum := 0
        for sum > 0 {
            newSum += sum % 10
            sum /= 10
        }
        sum = newSum
        k--
    }
    return sum
}

File: C++/sum-of-digits-of-string-after-convert.cpp
class Solution {
public:
    int getLucky(string s, int k) {
        int sum = 0;
        for (auto c : s) {
            int value = c - 'a' + 1;
            sum += value % 10 + value / 10;
        }
        while (k > 1) {
            int newSum = 0;
            while (sum > 0) {
                newSum += sum % 10;
                sum /= 10;
            }
            sum = newSum;
            k--;
        }
        return sum;
    }
};